docs/developer-docs/latest/update-migration-guides/migration-guides/v4/plugin/update-folder-structure.md

```diff
@@ -0,0 +1,362 @@
+---
+title: v4 Plugin Migration - Update folder structure - Strapi Developer Docs
+description:
+canonicalUrl:
+sidebarDepth: 2
+---
+
+<!-- TODO: update SEO -->
+
+# v4 plugin migration: Update the folder structure
+
+!!!include(developer-docs/latest/update-migration-guides/migration-guides/v4/snippets/plugin-migration-intro.md)!!!
+
+::: callout
+Strapi v3 plugins required a specific folder structure.
+
+In Strapi v4, plugins are developed using a programmatic API, which gives flexibility in the folder structure.
+:::
+
+The folder structure of a Strapi v4 plugin should meet the following requirements:
+
+- The root of the plugin folder should include:
+  - a `strapi-server.js` entry file, if the plugin interacts with Strapi's back-end (see [Server API](/developer-docs/latest/setup-deployment-guides/configurations/optional/environment.md))
+  - a `strapi-admin.js` entry file, if the plugin interacts with Strapi's admin panel (see [Admin Panel API](/developer-docs/latest/setup-deployment-guides/configurations/optional/environment.md)).
+
+- `strapi-admin.js` and `strapi-server.js` should export the plugin's interface.
+
+As long as these requirements are met, the rest of the folder structure is up to you.
+
+::: details Example of a Strapi v4 plugin structure
+
+```jsx
+my-plugin
```

Comment:
This is the folder structure our CLI generates but other instructions in this guide don't follow it :) We should decide our recommended structure

Reply:
The guide is intended to follow this structure since this is also the structure the codemod follows. @pwizla if you have any questions about paths that aren't clear in the blog article just ping me :)

Reply:
Marking this resolved as the guide does not mention the CLI and I agree we should stick to what codemods do. (but also see my other comments about these topics 😄 )

Reply:
The CLI and the codemods should be the same, at least I tried to build it that way